crates/bevy_ui/src/focus.rs

```diff
@@ -172,41 +179,43 @@ pub fn ui_focus_system(
                     }
                     None
                 }
-            },
-        )
-        .collect::<Vec<_>>();
-
-    moused_over_z_sorted_nodes.sort_by_key(|(_, _, _, z)| -*z);
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<Entity>>()
+        .into_iter();
 
-    let mut moused_over_z_sorted_nodes = moused_over_z_sorted_nodes.into_iter();
     // set Clicked or Hovered on top nodes
-    for (entity, focus_policy, interaction, _) in moused_over_z_sorted_nodes.by_ref() {
-        if let Some(mut interaction) = interaction {
+    let mut iter = node_query.iter_many_mut(moused_over_nodes.by_ref());
+    while let Some(node) = iter.fetch_next() {
+        if let Some(mut interaction) = node.interaction {
             if mouse_clicked {
                 // only consider nodes with Interaction "clickable"
                 if *interaction != Interaction::Clicked {
                     *interaction = Interaction::Clicked;
                     // if the mouse was simultaneously released, reset this Interaction in the next
                     // frame
                     if mouse_released {
-                        state.entities_to_reset.push(entity);
+                        state.entities_to_reset.push(node.entity);
                     }
                 }
             } else if *interaction == Interaction::None {
                 *interaction = Interaction::Hovered;
             }
         }
 
-        match focus_policy.cloned().unwrap_or(FocusPolicy::Block) {
+        match node.focus_policy.unwrap_or(&FocusPolicy::Block) {
             FocusPolicy::Block => {
                 break;
             }
             FocusPolicy::Pass => { /* allow the next node to be hovered/clicked */ }
         }
     }
     // reset lower nodes to None
-    for (_entity, _focus_policy, interaction, _) in moused_over_z_sorted_nodes {
-        if let Some(mut interaction) = interaction {
+    let mut iter = node_query.iter_many_mut(moused_over_nodes);
```

Comment:
This pattern needs a comment to explain why we're using it  + the underlying logic.

Reply:
Would you prefer if it used a simpler loop with `get_mut` instead? I changed this with the hope that it may improve performances, but it's unclear if it really does and maybe clarity would be better here? Otherwise I can add a simple comment, something along the line of:
```
// use the [`iter_many_mut`] pattern to go over remaining nodes. it allows iterating over a specific
// subset of entities in the node query but it doesn't support using it as a simple [`Iterator`].
```

Reply:
You can ignore my previous question. I added multiple new comments in the code to explain what's happening and why.

Reply:
LGTM now!